pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -141,6 +141,11 @@ message FunctionMetaData {
     uint64 version = 3;
     uint64 createTime = 4;
     map<int32, FunctionState> instanceStates = 5;
+    FunctionAuthenticationSpec functionAuthSpec = 6;
+}
+
+message FunctionAuthenticationSpec {
+    string data = 1;
```

Comment:
I am wondering why are you using `string`, not `bytes`. my concern of `string` is that encoding with locales make things very tricky and hard to debug.

Reply:
I didn't know what kind of params/data would need to be stored so I just based this off of the authParams String format we already have in Pulsar. We can change this to bytes.

Reply:
also you can always base64 encode a binary to be stored as a string

Reply:
@jerrypeng bytes is a more natural representation :-)

Reply:
@sijie i have changed